### PR TITLE
Test behavior of APPLICATION_JSON String return types

### PR DESCRIPTION
**Pull request by uschi2000**
This change is [Reviewable](https://reviewable.io/reviews/palantir/http-remoting/421)

### CODE REVIEW

File: jaxrs-clients/src/test/java/feign/TextDelegateDecoderTest.java [hunk 1]
```diff
@@ -78,6 +79,13 @@ public void testUsesStringDecoderWithTextPlain() throws Exception {
     }
 
     @Test
+    public void testCannotReturnStringWithMediaTypeJson() throws Exception {
+        assertThatThrownBy(() -> service.getJsonString("foo"))
```
Comment (uschi2000) [inline jaxrs-clients/src/test/java/feign/TextDelegateDecoderTest.java:83]: @markelliot we chatted about this 1-2 weeks ago and I claimed that `String` return type doesn't work with media type APPLICATION_JSON. is the behavior different in our internal app server?

Comment (markelliot) [reply jaxrs-clients/src/test/java/feign/TextDelegateDecoderTest.java:83]: the server is returning the raw string in this example, that behavior is the same.

Comment (markelliot) [reply jaxrs-clients/src/test/java/feign/TextDelegateDecoderTest.java:83]: (Feign throwing the exception means that the content `foo` came back over the wire)

**Comment by uschi2000** [comment]
Yep, I know.  You said that you had made it so that returning String is
possible with APPLICATION_JSON though?

On May 10, 2017 3:15 PM, "Mark Elliot" <notifications@github.com> wrote:

*@markelliot* commented on this pull request.
------------------------------

In jaxrs-clients/src/test/java/feign/TextDelegateDecoderTest.java
<https://github.com/palantir/http-remoting/pull/421#discussion_r115736148>:

> @@ -78,6 +79,13 @@ public void testUsesStringDecoderWithTextPlain() throws Exception {
     }

     @Test
+    public void testCannotReturnStringWithMediaTypeJson() throws Exception {
+        assertThatThrownBy(() -> service.getJsonString("foo"))

(Feign throwing the exception means that the content foo came back over the
wire)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/palantir/http-remoting/pull/421#discussion_r115736148>,
or mute the thread

**Pull request merged**